Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -58,9 +58,10 @@ class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor<Site>, SearchViewController
             inProgress = deferred as? Cancellable
 
             deferred.upon() { result in
-                DispatchQueue.main.async {
-                    self.inProgress = nil
+                guard let inProgress = self.inProgress, let deferred = deferred as? Cancellable, inProgress === deferred else {
+                    return
```

Comment:
I don't like this block actually. For correctness we should have:
```
    var isCancelled = false
    DispatchQueue.main.sync() { isCancelled = <the logic I have here to check >}
```

or an `func isCancelled() -> Success` and the rest of this function is in the upon() block for that.